quartz/plugins/transformers/ofm.ts

```diff
@@ -114,9 +114,14 @@ const commentRegex = new RegExp(/%%(.+)%%/, "g")
 // from https://github.com/escwxyz/remark-obsidian-callout/blob/main/src/index.ts
 const calloutRegex = new RegExp(/^\[\!(\w+)\]([+-]?)/)
 const calloutLineRegex = new RegExp(/^> *\[\!\w+\][+-]?.*$/, "gm")
-// (?:^| )   -> non-capturing group, tag should start be separated by a space or be the start of the line
-// #(\w+)    -> tag itself is # followed by a string of alpha-numeric characters
-const tagRegex = new RegExp(/(?:^| )#(\p{L}+)/, "gu")
+// (?:^| )                        -> non-capturing group, tag should start be separated by a space or be the start of the line
+// #                              -> tag itself is preceded by a hashtag
+// (\p{L}+                        -> followed by a string of (Unicode-aware) alpha-numeric characters
+// (?:[-_]\p{L}+)*)               -> an (optional) suffix string preceded by a hyphen "-" or underscore "_"
+const nestedTagRegex = new RegExp(
+  /(?:^| )#(\p{L}+(?:[-_]\p{L}+)*)(?:\/(\p{L}+(?:[-_]\p{L}+)*))?/,
```

Comment:
note that Obsidian supports multiple nesting levels (e.g. #a/b/c), we should account for that. Also, I think hyphen and underscore can appear anywhere in the tag (i.e. `#--` is a valid Obsidian tag) which should simplify the regex so that we can replace `\p{L}` with `[\p{L}-_]`

Reply:
ah yup, great points. will update!